test/suite/test_prepare_hs03.py

```diff
@@ -0,0 +1,183 @@
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from helper import copy_wiredtiger_home
+import unittest, wiredtiger, wttest
+from wtdataset import SimpleDataSet
+import os, shutil
+
+def timestamp_str(t):
+    return '%x' % t
+
+# test_prepare_hs03.py
+# test to ensure salvage, verify & simulating crash are working for prepared transactions.
+class test_prepare_hs03(wttest.WiredTigerTestCase):
+    # Force a small cache.
+    conn_config = 'cache_size=50MB'
+
+    # Create a small table.
+    uri = "table:test_prepare_hs03"
+
+    def corrupt_file(self):
+        filename="test_prepare_hs03.wt"
+        self.assertEquals(os.path.exists(filename), True)
+
+        with open(filename, 'r+') as log:
+            log.seek(1024)
+            log.write('Bad!' * 1024)
+
+    def corrupt_salvage_verify(self):
+        self.corrupt_file()
+        self.session.salvage(self.uri, "force")
+        self.session.verify(self.uri, None)
+
+    def prepare_updates(self, ds, nrows, nsessions, nkeys):
+        # Insert some records with commit timestamp, corrupt file and call salvage, verify before checkpoint.
+
+        # Commit some updates to get eviction and history store fired up
+        bigvalue1 = b"bbbbb" * 2
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(1, nkeys):
+            self.session.begin_transaction('isolation=snapshot')
+            tkey = 'C' + ds.key(nrows + i)
+            cursor.set_key(tkey)
+            cursor.set_value(bigvalue1)
+            self.assertEquals(cursor.insert(), 0)
+            self.session.commit_transaction('commit_timestamp=' + timestamp_str(1))
+        cursor.close()
+
+        # Corrupt the table, Call salvage to recover data from the corrupted table and call verify
+        self.corrupt_salvage_verify()
+
+        # Call checkpoint
+        self.session.checkpoint()
+
+        # Corrupt the table, Call salvage to recover data from the corrupted table and call verify
+        self.corrupt_salvage_verify()
+
+        # Have prepared updates in multiple sessions. This should ensure writing
+        # prepared updates to the history store
+        sessions = [0] * nsessions
+        cursors = [0] * nsessions
+        bigvalue2 = b"ccccc" * 2
+        for j in range (0, nsessions):
+            sessions[j] = self.conn.open_session()
+            sessions[j].begin_transaction('isolation=snapshot')
+            cursors[j] = sessions[j].open_cursor(self.uri)
+            # Each session will update many consecutive keys.
+            start = (j * nkeys)
+            end = start + nkeys
+            for i in range(start, end):
+                tkey = 'P' + ds.key(nrows + i)
```

Comment:
Similar to comment above, these keys won't match the original keys.

Reply:
Fixed